components/scream/src/physics/shoc/shoc_update_host_dse_impl.hpp

```diff
@@ -0,0 +1,40 @@
+#ifndef SHOC_UPDATE_HOST_DSE_IMPL_HPP
+#define SHOC_UPDATE_HOST_DSE_IMPL_HPP
+
+#include "shoc_functions.hpp" // for ETI only but harmless for GPU
+
+namespace scream {
+namespace shoc {
+
+template<typename S, typename D>
+KOKKOS_FUNCTION
+void Functions<S,D>
+::update_host_dse(
+  const MemberType& team,
+  const Int& nlev,
+  const uview_1d<const Spack>& thlm,
+  const uview_1d<const Spack>& shoc_ql,
+  const uview_1d<const Spack>& exner,
+  const uview_1d<const Spack>& zt_grid,
+  const Scalar& phis,
+  const uview_1d<Spack>& host_dse)
+{
+  const Int nlev_pack = ekat::pack::npack<Spack>(nlev);
+
+  // Constants used
+  const auto lcond = C::LatVap;
+  const auto cp = C::CP;
+  const auto ggr = C::gravit;
+
+  Kokkos::parallel_for(Kokkos::TeamThreadRange(team, nlev_pack), [&] (const Int& k) {
+      auto range_pack = ekat::pack::range<IntSmallPack>(k*Spack::n);
+
+      Spack temp = (thlm(k)+(lcond/cp)*shoc_ql(k))/exner(k);
+      host_dse(k).set(range_pack < nlev, cp*temp+ggr*zt_grid(k)+phis);
```

Comment:
For simple slot-level operations, you don't need to mask with range_pack < nlev. Arithmetic in unused slots is fine. We just have to mask when doing things like reductions, BCs, shifted indices, and scans, although even in these cases not always.